component/metrics/scraper/scraper.go

```diff
@@ -0,0 +1,237 @@
+package scraper
+
+// NOTE: This is a placeholder component for scraper for testing of the appendable, it should NOT be considered final
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/component/metrics"
+
+	"github.com/alecthomas/units"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+	"github.com/rfratto/gohcl"
+)
+
+func init() {
+	scrape.UserAgent = fmt.Sprintf("GrafanaAgent/%s", build.Version)
+
+	component.Register(component.Registration{
+		Name: "metrics.scraper",
+		Args: Config{},
+		Build: func(o component.Options, c component.Arguments) (component.Component, error) {
+			return NewComponent(o, c.(Config))
+		},
+	})
+	component.RegisterGoStruct("MetricsReceiver", metrics.Receiver{})
+}
+
+// Config represents the input state of the metrics_scraper component.
+type Config struct {
+	Targets []TargetGroup `hcl:"targets"`
+
+	HonorLabels           bool                `hcl:"honor_labels,optional"`
+	HonorTimestamps       bool                `hcl:"honor_timestamps,optional"`
+	Params                map[string][]string `hcl:"params,optional"`
+	ScrapeInterval        time.Duration       `hcl:"scrape_interval,optional"`
+	ScrapeTimeout         time.Duration       `hcl:"scrape_timeout,optional"`
+	MetricsPath           string              `hcl:"metrics_path,optional"`
+	Scheme                string              `hcl:"scheme,optional"`
+	BodySizeLimit         units.Base2Bytes    `hcl:"body_size_limit,optional"`
+	SampleLimit           uint                `hcl:"sample_limit,optional"`
+	TargetLimit           uint                `hcl:"target_limit,optional"`
+	LabelLimit            uint                `hcl:"label_limit,optional"`
+	LabelNameLengthLimit  uint                `hcl:"label_name_length_limit,optional"`
+	LabelValueLengthLimit uint                `hcl:"label_value_length_limit,optional"`
+
+	// TODO(rfratto): http client config
+	Receivers []*metrics.Receiver `hcl:"receivers"`
```

Comment:
There is some stuttering that goes on here with flow, in testing it got called with 0 receivers, then 1 and then 2 receivers. Not a big deal but does create some churn.